components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -3672,6 +3833,116 @@ private void updateUserClaims(User user, Map<String, String> oldClaimList,
+        /* Remove simple multi-valued claims URIS from existing claims and updated user's claims.
+           oldClaimList and newClaimList are not modifying to reuse for NotImplemented exception. */
+        Map<String, String> oldClaimListExcludingMultiValuedClaims = new HashMap<>(oldClaimList);
+        oldClaimListExcludingMultiValuedClaims.keySet().removeAll(allSimpleMultiValuedClaimsList.keySet());
+
+        Map<String, String> newClaimListExcludingMultiValuedClaims = new HashMap<>(newClaimList);
+        newClaimListExcludingMultiValuedClaims.keySet().removeAll(allSimpleMultiValuedClaimsList.keySet());
+
+        Map<String, String> userClaimsToBeAdded = new HashMap<>(newClaimListExcludingMultiValuedClaims);
+        Map<String, String> userClaimsToBeDeleted = new HashMap<>(oldClaimListExcludingMultiValuedClaims);
+        Map<String, String> userClaimsToBeModified = new HashMap<>();
+
+        // Get all the old claims, which are not available in the new claims.
+        userClaimsToBeDeleted.keySet().removeAll(newClaimListExcludingMultiValuedClaims.keySet());
+
+        // Get all the new claims, which are not available in the existing claims.
+        userClaimsToBeAdded.keySet().removeAll(oldClaimListExcludingMultiValuedClaims.keySet());
+
+        // Get all new claims, which are only modifying the value of an existing claim.
+        for (Map.Entry<String, String> eachNewClaim : newClaimListExcludingMultiValuedClaims.entrySet()) {
+            if (oldClaimListExcludingMultiValuedClaims.containsKey(eachNewClaim.getKey()) &&
+                    !oldClaimListExcludingMultiValuedClaims.get(eachNewClaim.getKey())
+                            .equals(eachNewClaim.getValue())) {
+                userClaimsToBeModified.put(eachNewClaim.getKey(), eachNewClaim.getValue());
+            }
+        }
+
+        // Remove user claims.
+        for (Map.Entry<String, String> entry : userClaimsToBeDeleted.entrySet()) {
+            if (!isImmutableClaim(entry.getKey())) {
+                carbonUM.deleteUserClaimValueWithID(user.getId(), entry.getKey(), null);
+            }
+        }
+
+        // Update user claims.
+        userClaimsToBeModified.putAll(userClaimsToBeAdded);
+        if (MapUtils.isEmpty(simpleMultiValuedClaimsToBeAdded) &&
+                MapUtils.isEmpty(simpleMultiValuedClaimsToBeRemoved)) {
+            // If no multi-valued attribute is modified.
+            carbonUM.setUserClaimValuesWithID(user.getId(), userClaimsToBeModified, null);
+        } else {
+            carbonUM.setUserClaimValuesWithID(user.getId(), convertClaimValuesToList(oldClaimList),
+                    simpleMultiValuedClaimsToBeAdded, simpleMultiValuedClaimsToBeRemoved,
+                    convertClaimValuesToList(userClaimsToBeModified), null);
+        }
+    }
+
+    /**
+     * Convert the claim values to list of strings. Main usage is for multi-valued claims.
```

Comment:
Shall we not mention about the usage of the method? WDYT

Reply:
addressed in https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/266/commits/d419b5d649f8bc9093b6c4143d0fa188cce5ad99